lib/src/components/NxCombobox/NxCombobox.tsx

```diff
@@ -0,0 +1,328 @@
+      dropdownRef = useRef<HTMLDivElement>(null),
+      inputRef = useRef<HTMLDivElement>(null),
+
+      [focusableBtnIndex, setFocusableBtnIndex] = useState<number | null>(null),
+      [showDropdown, setShowDropdown] = useState(false),
+      [inlineStyle, setInlineStyle] = useState(false),
+      inputVal = autoSelectedVal && inlineStyle ? autoSelectedVal : value,
+
+      inputId = useUniqueId('nx-combobox-input', id),
+      alertDropdownId = useUniqueId('nx-combobox-alert-dropdown'),
+      dropdownId = useUniqueId('nx-combobox-dropdown'),
+      dropdownBtnId = useUniqueId('nx-dropdown-button'),
+      focusableBtnId = focusableBtnIndex !== null ?
+        dropdownRef.current?.children[focusableBtnIndex].id : undefined,
+
+      className = classnames('nx-combobox', classNameProp, {
+        'nx-combobox--dropdown-showable': showDropdown
+      }),
+      alertClassName = classnames('nx-combobox__alert', {
+        'nx-combobox__alert--error': !!loadError
+      }),
+      inputDescribedby = classnames(ariaDescribedBy, {
+        [`${alertDropdownId}`]: isAlert && showDropdown
+      });
+
+  // There is a requirement that when there is an error querying the data, if the user navigates away from
+  // the component and then comes back to it the search should be retried automatically
+  function handleComponentFocus(evt: FocusEvent<HTMLDivElement>) {
+    if (loadError) {
+      // check if this is focus coming into the component from somewhere else on the page, not just moving between
+      // children of this component and not from focus coming into the browser from some other window
+      const comingFromOutsidePage = evt.relatedTarget === null,
+          comingFromChildNode = evt.relatedTarget instanceof Node && evt.currentTarget.contains(evt.relatedTarget);
+
+      if (!(comingFromOutsidePage || comingFromChildNode)) {
+        doSearch(value);
+      }
+      else {
+        setShowDropdown(true);
+      }
+    }
+  }
+
+  function handleComponentBlur(evt: FocusEvent<HTMLDivElement>) {
+    setFocusableBtnIndex(null);
+
+    // Check if useragent string contains Safari
+    const isSafari = navigator.userAgent.indexOf('Safari') !== -1 && navigator.userAgent.indexOf('Chrome') === -1;
+
+    // Check if the new focused element is a child of the parent, if not, then close the dropdown menu
+    // For the issue that clicking the button in Safari does not focus it, we decide to show the dropdown in Safari
+    // when blur.
+    if (!(evt.relatedTarget instanceof Node && evt.currentTarget.contains(evt.relatedTarget)) && !isSafari) {
+
+      // The automatically selected suggestion becomes the value of the combobox
+      // when the combobox loses focus.
+      if (autoComplete) {
+        if (autoSelectedVal && inlineStyle) {
+          onChange(autoSelectedVal);
+          // sync the dropdown
+          onSearch(autoSelectedVal);
+        }
+        else {
+          onSearch(value);
```

Comment:
What's the purpose of this particular `onSearch` call?

Reply:
This was to sync the dropdown when there is value typed in the input box but then tab away without making selections. However this is no longer needed after removing the autoSelectedVal. Thanks.